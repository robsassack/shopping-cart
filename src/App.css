* {
  padding: 0;
  margin: 0;
  font-family: "Roboto", Helvetica, Arial, sans-serif;
}

button {
  cursor: pointer;
  background-color: #1a1a1a;
  color: white;
  border: 1px solid #1a1a1a;
  border-radius: 5px;
  padding: 5px 10px;
  font-size: 1rem;
}

button:hover {
  background-color: #333;
  border: 1px solid #333;
  transform: scale(1.05);
}

button:active {
  transform: scale(0.95);
  background-color: #111;
}

.App {
  display: flex;
  flex-direction: column;
  height: 100vh;
}

.App--navbar {
  background-color: #1a1a1a;
  font-size: 1.5rem;
  box-shadow: 0 0 10px 0 rgba(0, 0, 0, 0.5);
  padding: 15px 40px;
  margin-bottom: 20px;
}

.App--navbar ul {
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  justify-content: space-between;
  gap: 20px;
  list-style: none;
}

.App--links {
  display: flex;
  flex-wrap: wrap;
  gap: 40px;
  align-items: center;
}

.App--navbar a {
  color: white;
  text-decoration: none;
}

.App--footer {
  display: flex;
  justify-content: space-between;
  text-align: center;
  margin-top: auto;
  background-color: #1a1a1a;
  color: white;
  font-size: 1.2rem;
  padding: 15px 40px;
}

.App--footer-repo {
  color: white;
  font-size: 1.3em;
}

.home {
  display: flex;
  flex-wrap: wrap;
  gap: 30px;
  margin: 20px;
  justify-content: center;
  align-items: center;
}

.home--image {
  max-width: 400px;
  width: 100%;
  border-radius: 20px;
}

.home--text {
  display: flex;
  flex-direction: column;
  gap: 15px;
  align-items: center;
}

.home--button {
  padding: 10px 20px;
  font-size: 1.2rem;
}

.shop {
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 20px;
  margin-bottom: 40px;
}

.shop--items-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  gap: 10px;
  max-width: 1000px;
}

.shop--items-container a {
  color: black;
  text-decoration: none;
}

.shop--item {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  gap: 5px;
  width: 200px;
  border: 1px solid grey;
  border-radius: 5px;
  padding: 10px 5px;
  text-align: center;
}

.shop--item:hover {
  box-shadow: 0 0 10px 0 rgba(0, 0, 0, 0.5);
  color: blue;
  transform: scale(1.05);
}

.shop--item:active {
  transform: scale(0.95);
}

.shop--img-container {
  display: flex;
  align-items: center;
  justify-content: center;
  height: 150px;
  width: 150px;
}

.shop--item img {
  max-width: 100%;
  max-height: 100%;
}

.item {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  align-items: center;
  margin-top: 40px;
}

.item--info {
  display: flex;
  flex-direction: column;
  gap: 15px;
  max-width: 500px;
  padding: 20px;
}

.item--info p {
  font-size: 1.5em;
}

.item--info button {
  padding: 10px 20px;
}

.item--image {
  max-height: 300px;
  max-width: 350px;
}

.cart {
  display: flex;
  flex-direction: column;
  gap: 20px;
  align-items: center;
  margin-bottom: 20px;
}

.cart--item {
  display: flex;
  align-items: center;
  max-width: 450px;
  border-bottom: 1px solid rgb(204, 204, 204);
  padding: 10px 0;
  margin: 0 30px;
}

.cart--image {
  max-height: 80px;
  max-width: 80px;
}

.cart--name {
  font-size: 1.2em;
}

.cart--info {
  display: flex;
  flex-direction: column;
  gap: 10px;
  padding: 10px;
}

.cart--quantity {
  display: flex;
  align-items: center;
  gap: 10px;
}

.cart--delete {
  background-color: rgb(204, 0, 0) !important;
}

.cart--delete:hover {
  background-color: rgb(255, 0, 0) !important;
}

.cart--delete:active {
  background-color: rgb(153, 0, 0) !important;
}

.cart--checkout {
  background-color: rgb(0, 204, 0) !important;
}

.cart--checkout:hover {
  background-color: rgb(0, 255, 0) !important;
}

.cart--checkout:active {
  background-color: rgb(0, 153, 0) !important;
}

.cart--empty {
  font-size: 1.5em;
}
