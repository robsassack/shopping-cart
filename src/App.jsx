import { NavLink, Routes, Route } from "react-router-dom";
import Home from "./pages/Home";
import Shop from "./pages/Shop";
import Cart from "./pages/Cart";
import Item from "./pages/Item";
import NotFound from "./pages/NotFound";
import "./App.css";

function App() {
  return (
    <div className='App'>
      <nav>
        <ul>
          <li>
            <NavLink to='/'>Computer World</NavLink>
          </li>
          <li>
            <NavLink to='/shop'>Shop</NavLink>
          </li>
          <li>
            <NavLink to='/cart'>Cart</NavLink>
          </li>
        </ul>
      </nav>
      <Routes>
        <Route path='/' element={<Home />} />
        <Route path='/shop' element={<Shop />} />
        <Route path='/cart' element={<Cart />} />
        <Route path='/shop/:type/:id' element={<Item />} />
        <Route path='*' element={<NotFound />} />
      </Routes>
    </div>
  );
}

export default App;
