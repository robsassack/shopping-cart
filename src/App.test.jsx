import React from "react";
import { describe, expect, it } from "vitest";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { BrowserRouter } from "react-router-dom";
import App from "./App";

describe("App", () => {
  it("render app", () => {
    const { container } = render(<App />, { wrapper: BrowserRouter });
    expect(container).toMatchSnapshot();
  });

  it("render shop page", async () => {
    render(<App />, { wrapper: BrowserRouter });
    await userEvent.click(screen.getByRole("link", { name: /Go to Shop/i }));
    expect(screen.getByRole("heading").textContent).toMatch(/Shop Page/i);
  });

  it("render item page", async () => {
    render(<App />, { wrapper: BrowserRouter });
    await userEvent.click(
      screen.getByRole("link", { name: /Intel Core i9-13900K/i })
    );
    expect(screen.getByRole("heading").textContent).toMatch(
      /Intel Core i9-13900K/i
    );
  });

  it("add item to cart", async () => {
    const { container } = render(<App />, { wrapper: BrowserRouter });
    await userEvent.click(screen.getByRole("button", { name: /Add to Cart/i }));
    expect(container.querySelector(".cart--quantity").textContent).toContain(
      "1"
    );
    expect(container.querySelector(".cart--total").textContent).toMatch(
      "Total: $589.99"
    );
  });

  it("increase quantity of item in cart", async () => {
    const { container } = render(<App />, { wrapper: BrowserRouter });
    await userEvent.click(screen.getByRole("link", { name: /Shop/i }));
    await userEvent.click(
      screen.getByRole("link", { name: /Intel Core i9-13900K/i })
    );
    await userEvent.click(screen.getByRole("button", { name: /Add to Cart/i }));
    await userEvent.click(screen.getByRole("button", { name: /\+/i }));
    expect(container.querySelector(".cart--quantity").textContent).toContain(
      "2"
    );
    expect(container.querySelector(".cart--total").textContent).toMatch(
      "Total: $1179.98"
    );
  });

  it("decrease quantity of item in cart", async () => {
    const { container } = render(<App />, { wrapper: BrowserRouter });
    await userEvent.click(screen.getByRole("link", { name: /Shop/i }));
    await userEvent.click(
      screen.getByRole("link", { name: /Intel Core i9-13900K/i })
    );
    await userEvent.click(screen.getByRole("button", { name: /Add to Cart/i }));
    await userEvent.click(screen.getByRole("button", { name: /\+/i }));
    await userEvent.click(screen.getByRole("button", { name: /\-/i }));
    expect(container.querySelector(".cart--quantity").textContent).toContain(
      "1"
    );
    expect(container.querySelector(".cart--total").textContent).toMatch(
      "Total: $589.99"
    );
  });

  it("add two different items to cart", async () => {
    const { container } = render(<App />, { wrapper: BrowserRouter });
    await userEvent.click(screen.getByRole("link", { name: /Shop/i }));
    await userEvent.click(
      screen.getByRole("link", { name: /Intel Core i9-13900K/i })
    );
    await userEvent.click(screen.getByRole("button", { name: /Add to Cart/i }));
    await userEvent.click(screen.getByRole("link", { name: /Shop/i }));
    await userEvent.click(
      screen.getByRole("link", { name: /Nvidia GeForce RTX 3090/i })
    );
    await userEvent.click(screen.getByRole("button", { name: /Add to Cart/i }));
    expect(container.querySelector(".App--cart-link").textContent).toContain(
      "2"
    );
  });

  it("remove all items from cart", async () => {
    const { container } = render(<App />, { wrapper: BrowserRouter });
    await userEvent.click(screen.getByRole("link", { name: /Shop/i }));
    await userEvent.click(
      screen.getByRole("link", { name: /Intel Core i9-13900K/i })
    );
    await userEvent.click(screen.getByRole("button", { name: /Add to Cart/i }));
    await userEvent.click(screen.getByRole("link", { name: /Shop/i }));
    await userEvent.click(
      screen.getByRole("link", { name: /Nvidia GeForce RTX 3090/i })
    );
    await userEvent.click(screen.getByRole("button", { name: /Add to Cart/i }));
    const removeButtons = screen.getAllByRole("button", { name: /X/i });
    await userEvent.click(removeButtons[0]);
    await userEvent.click(removeButtons[1]);
    expect(container.querySelector(".App--cart-link").textContent).toContain(
      "0"
    );
    expect(container.querySelector(".cart--empty").textContent).toMatch(
      /Your cart is empty/i
    );
  });
});
