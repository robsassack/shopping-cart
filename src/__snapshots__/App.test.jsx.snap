// Vitest Snapshot v1

exports[`App > render app 1`] = `
<div>
  <div
    class="App"
  >
    <nav
      class="App--navbar"
    >
      <ul>
        <div
          class="App--logo"
        >
          <li>
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              <i
                class="fa-solid fa-globe"
              />
               Computer World
            </a>
          </li>
        </div>
        <div
          class="App--links"
        >
          <li>
            <a
              class=""
              href="/shop"
            >
              Shop
            </a>
          </li>
          <li
            class="App--cart-link"
          >
            <a
              class=""
              href="/cart"
            >
              <i
                class="fa-solid fa-cart-shopping"
              />
               (
              0
              )
            </a>
          </li>
        </div>
      </ul>
    </nav>
    <div
      class="home"
    >
      <img
        alt="Photo of Earth"
        class="home--image"
        src="/shopping-cart/src/assets/nasa-Q1p7bh3SHj8-unsplash.jpg"
      />
      <div
        class="home--text"
      >
        <h1>
          Welcome to Computer World!
        </h1>
        <p>
          All of the newest computer parts for the best prices!
        </p>
        <a
          class=""
          href="/shop"
        >
          <button
            class="home--button"
          >
            Go to Shop
          </button>
        </a>
      </div>
    </div>
  </div>
</div>
`;
